extensions/amp-iframe/0.1/amp-iframe.js

```diff
@@ -486,6 +488,36 @@ export class AmpIframe extends AMP.BaseElement {
     });
   }
 
+  /**
+   * Listen for Pym.js messages for 'height' and 'width'.
+   *
+   * @see http://blog.apps.npr.org/pym.js/
+   * @param {MessageEvent} event
+   * @private
+   */
+  listenForPymMessage_(event) {
+    if (
+      !this.iframe_ ||
+      event.source !== this.iframe_.contentWindow ||
+      typeof event.data !== 'string' ||
+      'pym' !== event.data.substr(0, 3)
+    ) {
+      return;
+    }
+
+    // The format of the message takes the form of `pym${id}xPYMx${type}xPYMx${message}`.
+    // The id is unnecessary for integration with amp-iframe; the possible types include
+    // 'height', 'width', 'parentPositionInfo', 'navigateTo', and  'scrollToChildPos'.
+    // Only the 'height' and 'width' messages are currently supported.
+    // See <https://github.com/nprapps/pym.js/blob/57feb68/src/pym.js#L85-L102>
+    const args = event.data.split(/xPYMx/).splice(2);
+    if ('height' === args[0]) {
+      this.updateSize_(parseInt(args[1], 10), undefined);
+    } else if ('width' === args[0]) {
+      this.updateSize_(undefined, parseInt(args[1], 10));
+    }
```

Comment:
Perhaps a message should be logged out for other message types so that a developer doesn't get confused about why it is not working.

Reply:
Sure. We could do a `user().warn()` for this.

Reply:
Done in 197c11002cd9ae4902dcbf6ac0500c6f2c621bc3